src/gsy_e/models/area/__init__.py

```diff
@@ -224,6 +238,10 @@ def update_descendants_strategy_prices(self):
             log.exception("area.update_descendants_strategy_prices failed.")
             return
 
+    def cycle_markets(self, current_time_slot: str):
+        """Perform operations that should be executed on market cycle."""
+        self._current_time_slot = current_time_slot
```

Comment:
You are assigning a `str` to `self._current_time_slot`, but the `now` property is supposed to return a `DateTime` object, which is needed for the `area.strategy.energy_rate.get(area.now, None))` lookup in the endpoint_buffer. AFAIU, you should pass a `DateTime` object to this method here in simulation.py

Reply:
You are correct, this indeed needs to be a datetime object. Fixing it now thanks!